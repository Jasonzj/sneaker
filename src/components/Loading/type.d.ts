export type Props = {
  spinning: boolean
  isShowText?: boolean
  backGroundColor?: string
}
