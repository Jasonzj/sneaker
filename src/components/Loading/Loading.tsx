import React from 'react'
import classNames from 'classnames'
import style from './loading.module.css'
import { Props } from './type'

const Loading: React.FC<Props> = ({ spinning, isShowText = false, backGroundColor }) => {
  return (
    <div
      className={classNames(style.loader, {
        [style.hidden]: !spinning,
        [style.fullScreen]: false,
      })}
      style={{ backgroundColor: backGroundColor }}
      title='loader'
    >
      <div className={style.wrapper}>
        <div className={style.inner} />
        {isShowText && <div className={style.text}>LOADING</div>}
      </div>
    </div>
  )
}

export default React.memo(Loading)
