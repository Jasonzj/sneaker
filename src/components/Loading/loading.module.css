.loader {
  display: block;
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  bottom: 0;
  z-index: 998;
  display: flex;
  justify-content: center;
  align-items: center;
  opacity: 1;
  text-align: center;
}
.loader.fullScreen {
  position: fixed;
}
.loader .wrapper {
  width: 100px;
  height: 100px;
  display: inline-flex;
  flex-direction: column;
  justify-content: space-around;
}
.loader .inner {
  width: 44px;
  height: 44px;
  margin: 0 auto;
  text-indent: -12345px;
  border-top: 4px solid rgba(0, 0, 0, 0.08);
  border-right: 4px solid rgba(0, 0, 0, 0.08);
  border-bottom: 4px solid rgba(0, 0, 0, 0.08);
  border-left: 4px solid rgba(0, 0, 0, 0.7);
  border-radius: 50%;
  z-index: 999;
}
.loader .inner :local {
  animation: spinner 600ms infinite linear;
}
.loader .text {
  width: 100px;
  height: 20px;
  text-align: center;
  font-size: 18px;
  letter-spacing: 4px;
  color: #000;
}
.loader.hidden {
  display: none;
  opacity: 0;
  transition: opacity 0.2s ease 0.1s, display 0.1s ease 1.5s;
}

.backGround {
  background-color: #ffffff;
}

@keyframes spinner {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
