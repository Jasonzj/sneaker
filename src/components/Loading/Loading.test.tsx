import React from 'react'
import { render, screen } from '@testing-library/react'
import Loading from './'

const props_true = {
  spinning: true,
  isShowText: true,
  backGroundColor: '#ffffff',
}

const props_false = {
  spinning: false,
  isShowText: false,
}

const setup = () => {
  const { rerender, asFragment } = render(<Loading {...props_true} />)
  const loader = screen.getByTitle('loader')

  return {
    loader,
    rerender,
    asFragment,
  }
}

test('should correct display loading text acording to props isShowText', () => {
  const { rerender, asFragment } = setup()

  expect(screen.getByText('LOADING')).toBeInTheDocument()
  expect(asFragment()).toMatchSnapshot()

  rerender(<Loading {...props_false} />)

  expect(screen.queryByText('LOADING')).not.toBeInTheDocument()
  expect(asFragment()).toMatchSnapshot()
})

test('should display specified backGroundColor', () => {
  const { rerender, loader } = setup()

  expect(loader).toHaveStyle('background-color: #ffffff')

  rerender(<Loading {...props_false} />)

  expect(loader).toHaveStyle('background-color: ')
})

test('should correct display or hidden acording to props spinning', () => {
  const { rerender, loader } = setup()

  expect(loader).not.toHaveClass('hidden')

  rerender(<Loading {...props_false} />)

  expect(loader).toHaveClass('hidden')
})
