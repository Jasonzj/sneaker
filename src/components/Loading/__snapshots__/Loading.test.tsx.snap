// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display correctly loading text acording to props isShowText 1`] = `
<DocumentFragment>
  <div
    class="loader"
    style="background-color: rgb(255, 255, 255);"
    title="loader"
  >
    <div
      class="wrapper"
    >
      <div
        class="inner"
      />
      <div
        class="text"
      >
        LOADING
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should display correctly loading text acording to props isShowText 2`] = `
<DocumentFragment>
  <div
    class="loader hidden"
    style=""
    title="loader"
  >
    <div
      class="wrapper"
    >
      <div
        class="inner"
      />
    </div>
  </div>
</DocumentFragment>
`;
